lib/Pusher.php

```diff
@@ -184,6 +184,37 @@ private function check_compatibility()
 		}
 
 	}
+
+	/**
+	 * validate number of channels and channel name format.
+	 */	
+	private function validate_channels($channels) {
+		if( count( $channels ) > 100 ) {
+			throw new PusherException('An event can be triggered on a maximum of 100 channels in a single call.');
+		}
+		
+		array_walk( $channels, array( $this, 'validate_channel' ) );
+	}
+
+	/**
+	 * Ensure a channel name is valid based on our spec
+	 */
+	private function validate_channel( $channel )
+	{
+		if ( ! preg_match( '/\A[-a-zA-Z0-9_=@,.;]+\z/', $channel ) ) {
+			throw new PusherException( 'Invalid channel name ' . $channel );
+		}
+	}
+
+	/**
+	 * Ensure a socket_id is valid based on our spec
+	 */
+	private function validate_socket_id( $socket_id )
+	{
+		if ( $socket_id && !preg_match( '/\A\d+\.\d+\z/', $socket_id ) ) {
```

Comment:
RE: https://github.com/pusher/pusher-http-php/pull/40#issuecomment-101588745

The idea here is that if `$socket_id` is any truthy value then we'll perform the regexp on it. If I'm honest I'm not sure what happens when `true` or `number` values are passed to `preg_match`.

Maybe the first check should be that it's of type `string`?

In any case some additional tests that make sure `null` values result in a socket_id being valid should probably be added.

Do you want to look at additional work on this, or should I?